cpp/STIR_PETSIRD_convertor.cpp

```diff
@@ -326,22 +326,43 @@ if (!stir::is_null_ptr(dynamic_cast<const stir::ProjDataInfoBlocksOnCylindrical
   // test
   {
     stir::CartesianCoordinate3D<float> coord_0, coord_1;
-    auto& pdi = dynamic_cast<const stir::ProjDataInfoCylindricalNoArcCorr &>(stir_proj_data_info);
-    for (unsigned int r=0; r< static_cast<unsigned>(stir_scanner->get_num_rings()); ++r)
-      for (unsigned int d=0; d < static_cast<unsigned>(stir_scanner->get_num_detectors_per_ring()); ++d)
-        {
-          pdi.find_cartesian_coordinates_given_scanner_coordinates(coord_0, coord_1, r, 0, d, 0, 1);
-          stir::DetectionPosition<> det_pos{d, r, 0};
-          const auto det_bin = get_PETSIRD_id_from_stir_det_pos(det_pos, stir_scanner);
-          const petsird::TypeOfModule type_of_module{0};
-          const auto expanded_detection_bin
-            = petsird_helpers::expand_detection_bin(scanner_info, type_of_module, det_bin);
-          const auto box_shape = petsird_helpers::geometry::get_detecting_box(scanner_info, type_of_module, expanded_detection_bin);
-          const auto mean_pos = mean_position(box_shape);
-          const auto p0 = stir::make_coordinate(mean_pos.c[2], -mean_pos.c[0], -mean_pos.c[1]);
-          const auto diff = coord_0 - p0;
-          std::cout << det_pos << coord_0 << p0 << diff << "\n";
-        }
+    if (!stir::is_null_ptr(dynamic_cast<const stir::ProjDataInfoBlocksOnCylindricalNoArcCorr *>(&stir_proj_data_info))){
+      auto& pdi = dynamic_cast<const stir::ProjDataInfoBlocksOnCylindricalNoArcCorr &>(stir_proj_data_info); // cast to blocks
```

Comment:
somewhat more concise is
```suggestion
    if (auto pdi_ptr = dynamic_cast<const stir::ProjDataInfoBlocksOnCylindricalNoArcCorr *>(&stir_proj_data_info)){
      auto& pdi = *pdi_ptr;
```

Reply:
OK, have changed to ProjDataInfoGenericNoArcCorr and make it more concise. Unfortunately, the pre-commit has made substantial changes to the file including code that I didn't change, is this OK?

<img width="1728" height="1117" alt="Screenshot 2025-11-12 at 11 18 03" src="https://github.com/user-attachments/assets/c3c30346-7818-42e0-8992-1013d8d6e8a8" />

Reply:
weird, but thanks for checking only the file in that you updated. I'll fix the rest later.